src/Npgsql/PgPassFile.cs

```diff
@@ -0,0 +1,71 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+
+namespace Npgsql
+{
+    /// <summary>
+    /// Represents a .pgpass file, which contains passwords for noninteractive connections
+    /// </summary>
+    class PgPassFile
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="PgPassFile"/> class
+        /// </summary>
+        /// <param name="fileName"></param>
+        public PgPassFile(string fileName)
+        {
+            FileName = fileName;
+        }
+
+        /// <summary>
+        /// File name being parsed for credentials
+        /// </summary>
+        public string FileName { get; }
+
+        /// <summary>
+        /// Parses file content and gets all credentials from the file
+        /// </summary>
+        /// <returns><see cref="IEnumerable{PgPassEntry}"/> corresponding to all lines in the .pgpass file</returns>
+        public IEnumerable<PgPassEntry> Entries
+        {
+            get
+            {
+                var serializedCredentials = File.ReadAllLines(FileName);
```

Comment:
Also, this whole property can be one simple LINQ expression, using an expression-bodied property.